### ISSUE TITLE
-[FCMemLogger addMessage:]

### ISSUE THREAD

**Issue by AtifSharaf** [freshdesk/freshchat-ios#73]
I am using SDK v2.9.0. Crash happened in foreground

EXC_BAD_ACCESS KERN_INVALID_ADDRESS 0x0000000971dae9b0

Stack trace

```
Crashed: com.apple.main-thread
0  libobjc.A.dylib                0x1f972f530 objc_msgSend + 16
1  App                       0x100947280 -[FCMemLogger addMessage:] + 4301845120
2  App                       0x10094739c -[FCMemLogger addErrorInfo:withMethodName:] + 4301845404
3  App                       0x1009bfb34 -[FCDataManager linkPSC:URL:mode:errorInfo:] + 4302338868
4  App                       0x1009bfe08 -[FCDataManager preparePSC] + 4302339592
5  App                       0x1009beeac +[FCDataManager sharedInstance] + 4302335660
6  App                       0x10096df68 +[FCUsers storeUserInfo:] + 4302004072
7  App                       0x1009c9744 -[Freshchat setUser:] + 4302378820
8  App                       0x10082cb0c specialized static Freshchat.setUser(userData:) + 4300688140 (FreshChatable.swift:201)
9  App                       0x1008143c4 specialized static SignedInUserManager.handleUserDataChange(userData:oldUserData:) + 4300587972 (SignedInUserManager.swift:141)
10 App                       0x100811970 closure #1 in closure #1 in SignedInUserManager.loadUserDataFromAPI() + 4300577136 (<compiler-generated>:4300577136)
11 App                       0x10087ed54 thunk for @escaping @callee_guaranteed () -> () + 4301024596 (<compiler-generated>:4301024596)
12 Foundation                     0x1fafa4cf4 <redacted> + 16
13 Foundation                     0x1faeb0d90 <redacted> + 72
14 Foundation                     0x1faeb0278 <redacted> + 740
15 Foundation                     0x1fafa6a78 <redacted> + 272
16 libdispatch.dylib              0x1f9f2b7f4 _dispatch_block_async_invoke2 + 104
17 libdispatch.dylib              0x1f9f7d7d4 _dispatch_client_callout + 16
18 libdispatch.dylib              0x1f9f2b008 _dispatch_main_queue_callback_4CF$VARIANT$mp + 1068
19 CoreFoundation                 0x1fa4d0b20 <redacted> + 12
20 CoreFoundation                 0x1fa4cba58 <redacted> + 1924
21 CoreFoundation                 0x1fa4cafb4 CFRunLoopRunSpecific + 436
22 GraphicsServices               0x1fc6cc79c GSEventRunModal + 104
23 UIKitCore                      0x226d2dc38 UIApplicationMain + 212
24 App                       0x1007da4f0 main + 4300350704 (ErrorLoadingCategoryCell.swift:27)
25 libdyld.dylib                  0x1f9f8e8e0 start + 4
```

**Comment by harish-kumarc** [comment]
Thanks @AtifSharaf for the logs, we are looking into it. Are you able to replicate it in dev envt or is it a production crash? Also if possible please send us the device info where its happening.
Thanks

**Comment by AtifSharaf** [comment]
@harish-kumarc this is a production crash. I am not able to reproduce it at my end.
included device info
<img width="803" alt="Screen Shot 2019-12-16 at 11 54 15 AM" src="https://user-images.githubusercontent.com/11500668/70888927-da1f7080-1ffa-11ea-88fe-12a9c9eef3b6.png">

**Comment by harish-kumarc** [comment]
@AtifSharaf , the crash is fixed with [v3.2.2](https://github.com/freshdesk/freshchat-ios/releases/tag/v3.2.2) Please update SDK and let us know if its fixed.

**Comment by harish-kumarc** [comment]
Closing the issue, Let us know if the issue resurface again.
